api/src/discord/listeners/interactionCreate.ts

```diff
@@ -0,0 +1,234 @@
+
+
+            } else if (interaction.customId.startsWith("archive-ctf-button-")) {
+                const ctfName = interaction.customId.replace("archive-ctf-button-", "");
+                await interaction.channel?.send(
+                    `Archiving the CTF channels and roles for ${ctfName}`
+                );
+                interaction.deferUpdate();
+
+                const categoryChannel = await interaction.guild?.channels.cache.find(
+                    (channel) =>
+                        channel.type === ChannelType.GuildCategory && channel.name === ctfName
+                ) as CategoryChannel;
+
+                const allMessages: any[] = []
+
+                interaction.guild?.channels.cache.map((channel) => {
+                    if (channel.type === ChannelType.GuildText && channel.parentId === categoryChannel.id) {
+                        fetchAllMessages(channel as TextBasedChannel).then(async (messages) => {
+                            allMessages.push(messages);
+
+                            // Wait until fetchAllMessages is completed before deleting the channels
+                            await channel.delete();
+                        });
+                    }
+                });
+
+                await categoryChannel.delete();
+
+                interaction.guild?.roles.cache.map((role) => {
+                    if (role.name === `${ctfName}`) {
+                        role.delete();
+                    }
+                });
+
+                interface Message {
+                    channel: string,
+                    content: string,
+                    author: string,
+                    timestamp: string
+                }
+
+                // put the archive in the archive channel of the ctf in the description
+                const niceMessages: string[] = allMessages.map((messages) => {
+                    let channelName = "";
+                    let niceMessage = "";
+
+                    messages = messages.reverse();
+
+                    if (messages.length > 0) {
+                        channelName = messages[0].channel;
+                        niceMessage += `## ${channelName}\n`;
+
+                        messages.forEach((message: Message) => {
+
+                            if (channelName != message.channel) {
+                                channelName = message.channel;
+                                niceMessage = `## ${channelName}\n`;
+                            }
+
+                            const timestamp = new Date(message.timestamp).toLocaleString();
+
+                            const formattedMessage = `[${timestamp}] ${message.author}: ${message.content}`;
+                            niceMessage += formattedMessage + "\n";
+
+
+                        });
+                    }
+
+                    return niceMessage;
+                });
+
+
+                const padUrl = await createPad(`${ctfName} archive`, niceMessages.join('\n'));
```

Comment:
What if it is over the maximum character limit of a pad?

Reply:
Maybe make a smart algorithm that splits the messages in half when it is over the limit and checks again (so keeps splitting until it is under the character limit)?